### PR TITLE
Remove test analyzers covered by nunit.analyzers (1st pass)

### PR DESCRIPTION
**Pull request by mthjones**
This removes the D2LTESTS001 and D2LTESTS002 diagnostics and their associated analyzers/codefixes. These are covered by the `nunit.analyzers`[^0] package we are bringing in instead.

[^0]: https://github.com/nunit/nunit.analyzers

### CODE REVIEW

File: src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs [hunk 1]
```diff
@@ -2,25 +2,6 @@
 
 namespace D2L.CodeStyle.TestAnalyzers.Common {
 	public static class Diagnostics {
-		public static readonly DiagnosticDescriptor TestCaseSourceStrings = new DiagnosticDescriptor(
-			id: "D2LTESTS001",
-			title: "Use nameof in TestCaseSource attributes.",
```
Comment (mthjones) [inline src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs:7]: https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit1002.md

File: src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs [hunk 2]
```diff
@@ -2,25 +2,6 @@
...
-			messageFormat: "String arguments in TestCaseSource not allowed. Use nameof( {0} ) instead.",
-			category: "Cleanliness",
-			defaultSeverity: DiagnosticSeverity.Error,
-			isEnabledByDefault: true,
-			description: "Using strings in TestCaseSource attributes creates false positives during dead code analysis. nameof should be used instead."
-		);
-
-		public static readonly DiagnosticDescriptor ValueSourceStrings = new DiagnosticDescriptor(
-			id: "D2LTESTS002",
-			title: "Use nameof in ValueSource attributes.",
```
Comment (mthjones) [inline src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs:17]: https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit1021.md

File: src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs (hunk 1 above)
Comment (omsmith) [reply src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs:7]: Warning vs Error matter for us?

**Review by j3parker — DISMISSED** [review]

File: src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs (hunk 1 above)
Comment (mthjones) [reply src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs:7]: We set `TreatWarningsAsErrors=true` globally, but we could up these ones to errors in the `.editorconfig` if we wanted to be more explicit.

File: src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs [hunk 3]
```diff
@@ -2,25 +2,6 @@
 
 namespace D2L.CodeStyle.TestAnalyzers.Common {
 	public static class Diagnostics {
-		public static readonly DiagnosticDescriptor TestCaseSourceStrings = new DiagnosticDescriptor(
```
Comment (omsmith) [inline src/D2L.CodeStyle.TestAnalyzers/Common/Diagnostics.cs:5]: Could you leave comments describing what these were? e.g.:

https://github.com/Brightspace/D2L.CodeStyle/blob/babbc3204f747d6543f107fe4c44f82f0f5f42d5/src/D2L.CodeStyle.Analyzers/Diagnostics.cs#L39-L41

**Review by omsmith — APPROVED** [review]

**Pull request merged**